File: LeaveManagementSystem.UI/src/app/interfaces/SubordinateDto.ts
import { SubordinateLeaveRequestDto } from './SubordinateleaveRequestDto';

export interface SubordinateDto {
  annualLeaveBalance: number;
  sickLeaveBalance: number;
  department: string;
  fullName: string;
  id: number;
  jobtitle: string;
  leaveRequestItemDtos: SubordinateLeaveRequestDto[];
}

File: LeaveManagementSystem.API/LeaveManagementSystem.API/Dtos/SubordinateDto.cs
namespace LeaveManagementSystem.API.Dtos
{
    public class SubordinateDto
    {
        public int Id { get; set; }
        public string FullName { get; set; }
        public string Jobtitle { get; set; }
        public byte AnnualLeaveBalance { get; set; }
        public byte SickLeaveBalance { get; set; }
        public string Department { get; set; }
        public List<SubordinateLeaveRequestDto> LeaveRequestItemDtos { get; set; } = new List<SubordinateLeaveRequestDto>();
    }
}
